src/MongoDB.Driver/CreateIndexOptions.cs

```diff
@@ -41,6 +41,7 @@ public class CreateIndexOptions
         private bool? _unique;
         private int? _version;
         private BsonDocument _weights;
+        private BsonDocument _wildcardProjection;
```

Comment:
Is BsonDocument the best we can do here?

Reply:
Moved this property to CreateIndexOptions<>Document> and replaced a type to `ProjectionDefinition<TDocument>`